main.py

```diff
@@ -1,21 +1,61 @@
+from __future__ import print_function
+
+import datetime
+
+import arcrest
+import dateutil.parser
+import dateutil.tz
+from arcrest.manageorg._community import Group
+from arcresthelper import common, orgtools, securityhandlerhelper
+
 import config
-from CanvasAPI import *
+import util
+from CanvasAPI import CanvasAPI
+
+runStartTime = datetime.datetime.now(tz=dateutil.tz.tzutc())
 
 
 def getCanvasInstance():
-    return CanvasAPI(config.Canvas.API_BASE_URL, authZToken=config.Canvas.API_AUTHZ_TOKEN)
+    return CanvasAPI(config.Canvas.API_BASE_URL,
+                     authZToken=config.Canvas.API_AUTHZ_TOKEN)
+
+
+def getArcGISConnection(securityinfo):
+    """
+    Get a connection object for ArcGIS based on configuration options
+    
+    :return: Connection object for the ArcGIS service
+    :rtype: arcresthelper.orgtools.orgtools
+    :raises: RuntimeError if ArcGIS connection is not valid
+    :raises: arcresthelper.common.ArcRestHelperError for ArcREST-related errors
+    """
+
+    if not isinstance(securityinfo, dict):
+        raise TypeError('Argument securityinfo type should be dict')
+
+    try:
+        arcGIS = securityhandlerhelper.securityhandlerhelper(securityinfo)
+    except common.ArcRestHelperError:
+        raise
```

Comment:
In a shell, run this:
```sh
python -c 'help("raise")' | head -10
```

You'll get:

```text
The "raise" statement
*********************

   raise_stmt ::= "raise" [expression ["," expression ["," expression]]]

If no expressions are present, "raise" re-raises the last exception
that was active in the current scope.  If no exception is active in
the current scope, a "TypeError" exception is raised indicating that
this is an error (if running under IDLE, a "Queue.Empty" exception is
raised instead).
```

In this case, I wanted to catch any `ArcRestHelperError` and re-raise it as the same type.  It will look like the exception came from my method.  On the other hand, since exceptions of other types aren't caught here, any that occur will continue to raise through the stack and appear to come from their original source.